lib/extensions/ar_virtual.rb

```diff
@@ -143,6 +143,7 @@ def define_delegate(method_name, method, to: nil, allow_nil: nil, default: nil)
       if allow_nil
         method_def = [
           "def #{method_name}(#{definition})",
+          "return self[:#{method_name}] if has_attribute?(:#{method_name})",
```

Comment:
Cool. thanks for tracking this down

Line numbers are tricky. the backtrace includes both blocks and method calls. Which is the reason for the `caller_locations(2, 1)` vs the expected `caller_locations(1,1)` at the beginning of the method.

We are basically doing what the original code did. Converting the heredoc into a single line:

```ruby
method_def = method_def.split("\n").map(&:strip).join ';'
```

Any variance from active support delegation is building up debt.
But we need to get rid of these N+1 queries. And making our models roll with the punches around virtual attributes is necessitating smarter delegation